### ISSUE TITLE
Gluon Routes its Packets to Wrong IPv6 Default Routers

### ISSUE THREAD

**Issue by lemoer** [freifunkh/ansible#155]
Gluon has a bug, that it routes its packets to the wrong default router.

I created an upstream issue here: [Link](https://github.com/freifunk-gluon/gluon/issues/2180). A good explanation of the problem is given there.

I suspect, this may take a while to be fixed. Therefore we should fix this downstream by applying the suggested workaround. As it became late now, 

i just hacked this manually in `/etc/systemd/network/10-gre-leintor.network` on sn01:
```
[Route]
Destination=::/0
Source=2a02:790:ff::/48
Gateway=fe80::200:5efe:5103:65e
Table=42
```

Packets of the foreign prefix are now routed via `gre-leintor` to the correct exit node. That's where they should have been routed to in the first place by the gluon nodes. 

We should add a corresponding config on leintor aswell and add it to ansible.

**Comment by lemoer** [comment]
I added the milestone "Beginn der stabilen Phase", as this is marked as bug and should therefore fixed before the "stabile Phase".

**Comment by AiyionPrime** [comment]
You've got quite a bunch of issues assigned to you; this is not.
Please tell, should all supernodes behave like this? Or ist this just superexitnode related?

And what should be the pendant configuration on leintor for this?
Changes in `/etc/systemd/network/10-gre-sn01.network`?
Are they implemented yet and I just overlook them?

**Comment by lemoer** [comment]
Thanks for picking this up.

The idea is that we "cross-route" packets, if necessary. We have two public
IPv6 prefixes for Freifunk Hannover. One is supposed to be "ausgeleitet" to
the internet by leintor and one is supposed to be "ausgeleitet" by "sn01".
However, sometimes packets which should be "ausgeleitet" at sn01 arrive at
sn05, sn09 or sn10 and will therefore be "ausgeleitet" at leintor.
Sometimes packets, which should be "ausgeleitet" at leintor arrive at
sn01.This will not work in all cases.

Therefore sn05, sn09 and sn10 should better decide to route those packages
into a "gre-sn01" tunnel. They can decide this based on the source IP of
the source IP of the packet by adding a route with `Source=...`. This will
then only apply to packets with that particular source. This is only a
problem for packets, which are "ausgeleitet".

 The same goes for sn01. Sn01 should route packets to leintor, if they are
supposed to be ausgeleitet by leintor (by their source ip).

So to sum up, we need to...:
1. ... add a route on sn01 towards leintor (based on source).
2. ... add a route on sn05, sn09 and sn10 towards sn01 (based on source).

Alternatively instead of 2., we could let sn05, sn09 and sn10 let route
packets first to leintor (as they would normally do) and install a source
specific route on leintor then. This would also direct packets to the
correct destination. This one might be easier as gre-tunnels tunnels are
existing and it has to be installed only at one place.

Regards,
Leo

**Comment by lemoer** [comment]
In case my explainations were not clear enough, you can contact me via
phone :)

**Comment by AiyionPrime** [comment]
Will look into this after the exam; eta 18:00 will be in mumble.

**Comment by lemoer** [comment]
Currently it's still set manually in /etc/. However, as nobody should roll anything in the infrastructure freeze week anyways, this should not brick in this week. Therefore I'll remove it from the milestone.

**Comment by lemoer** [comment]
This is partially fixed now by 66d14a8703681df1dc9f2b0c63e0e55dd88375e6 (#189).

Before 66d14a8703681df1dc9f2b0c63e0e55dd88375e6 the situation was the following:
1. sn01 routed packets originating from `2001:678:978::/48` to the internet.
2. sn01 routed packets originating from `2a02:790::/48` to the internet.
3. leintor routes packets originating from `2a02:790::/48` to the internet.
4. leintor routes packets originating from `2001:678:978::/48` to the internet.

Cases 2. and 4. are theoretically bad. Exitnodes should only route their own prefixes to the internet. However we only observed problems with case 2. Case 4. seems to be fine. I guess upstreams at cix are dropping bogus packets, but hostway does not.

With 66d14a8703681df1dc9f2b0c63e0e55dd88375e6 the situation is currently:
1. sn01 routes packets originating from `2001:678:978::/48` to the internet.
2. sn01 routes packets originating from `2a02:790::/48` to leintor.
3. leintor routes packets originating from `2a02:790::/48` to the internet.
4. leintor routes packets originating from `2001:678:978::/48` to the internet.

This means:
- Case 2. is now handled correctly.
- Case 4. is still not handled correctly.
- But case 4. doesn't seem to be too critical (as described above). Hostway seems to tolerate this.
- If someone wants to pick this up, he just needs to create a `gre-sn01` interface on sn05, sn09 and sn10. Then things should work out magically.

**Comment by lemoer** [comment]
Case 4 is handled correctly also, since sn09 and sn10 should route to clevertor.
